devito/function.py

```diff
@@ -669,19 +653,27 @@ def laplace(self):
         derivative wrt. all spatial dimensions.
         """
         derivs = tuple('d%s2' % d.name for d in self.space_dimensions)
-
         return sum([getattr(self, d) for d in derivs[:self.ndim]])
 
     def laplace2(self, weight=1):
         """
         Generates a symbolic expression for the double Laplacian
         wrt. all spatial dimensions.
         """
-        order = self.space_order/2
-        first = sum([second_derivative(self, dim=d, order=order)
-                     for d in self.space_dimensions])
-        return sum([second_derivative(first * weight, dim=d, order=order)
-                    for d in self.space_dimensions])
+        derivs = tuple('d%s2' % d.name for d in self.space_dimensions)
+        return sum([getattr(self.laplace * weight, d) for d in derivs[:self.ndim]])
+
+    def getdiff(self, name):
+        """
+        Returns finite dofference aproximation 'name' if depends on that dimension
+        or 0
+        """
+        if name in self.derivatives:
+            return getattr(self, name)
+        else:
+            warning("FD shortcut %s not found for Function %s " % (name, self) +
```

Comment:
warning or raise an error?

Reply:
warning, assuming if don't exist doesn't depend on it so returns 0